tools/src/main/java/org/apache/kafka/tools/consumer/group/ConsumerGroupCommand.java

```diff
@@ -1024,6 +1028,21 @@ private Map<TopicPartition, OffsetAndMetadata> prepareOffsetsToReset(String grou
             return null;
         }
 
+        private void checkAllTopicPartitionsHaveLeader(Collection<TopicPartition> partitionsToReset) {
+            List<TopicPartition> partitionsWithoutLeader = filterNoneLeaderPartitions(partitionsToReset);
+            if (!partitionsWithoutLeader.isEmpty()) {
+                // append the TopicPartition list string
```

Comment:
nit: You could do this string concatenation much more neatly with something like
```
String partitionStr = partitionsWithoutLeader.stream().map(TopicPartition::toString).collect(Collectors.joining(","));
```